app/src/main/java/com/celzero/bravedns/service/BraveVPNService.kt

```diff
@@ -483,23 +515,17 @@ class BraveVPNService : VpnService(), ConnectionCapabilityMonitor.NetworkListene
                 }
             }
 
-            if (appMode.getDNSType() == 3) {
-                try {
-                    //For DNS proxy mode, if any app is set then exclude the application from the list
-                    val dnsProxyEndpoint = dnsProxyEndpointRepository.getConnectedProxy()
-                    if (!dnsProxyEndpoint.proxyAppName.isNullOrEmpty() && dnsProxyEndpoint.proxyAppName != "Nobody") {
-                        Log.i(LOG_TAG, "$FILE_LOG_TAG DNS Proxy mode is set with the app - ${dnsProxyEndpoint.proxyAppName!!}- added to excluded list")
-                        builder = builder.addDisallowedApplication(dnsProxyEndpoint.proxyAppName!!)
-                    } else {
-                        Log.i(LOG_TAG, "$FILE_LOG_TAG DNS Proxy mode is set with the app - ${dnsProxyEndpoint.proxyAppName!!}- added to excluded list")
-                    }
-                }catch (e : Exception){
-                    Log.w(LOG_TAG,"Exception while excluding the proxy app from VPN")
+            if (appMode.getDNSType() == PREF_DNS_MODE_PROXY) {
+                //For DNS proxy mode, if any app is set then exclude the application from the list
+                val dnsProxyEndpoint = dnsProxyEndpointRepository.getConnectedProxy()
+                Log.i(LOG_TAG_VPN, "DNS Proxy mode is set with the app name as ${dnsProxyEndpoint.proxyAppName}")
+                if (!dnsProxyEndpoint.proxyAppName.isNullOrEmpty() && dnsProxyEndpoint.proxyAppName != getString(R.string.settings_app_list_default_app)) {
```

Comment:
Created a new method `isExcludePossible()`. This will check whether the VPN is in lockdown mode, if yes then will show a toast to the user else will disallow the application from the builder. 
The similar is added to Orbot and SOCKS5 proxy. 
String- Excluding %1$s for %2$s in lockdown mode can cause Internet connectivity issues
%1$s - app name
%2$s - DNSProxy / SOCKS5 Proxy / Tor-as-a-proxy